build/template-build-and-run-all-tests.yaml

```diff
@@ -93,4 +93,37 @@ jobs: #Build and stage projects
       Codeql.SkipTaskAutoInjection: true
 
   steps:
-  - template: template-test-on-linux.yaml
\ No newline at end of file
+  - template: template-test-on-linux.yaml
+  
+- job: 'RunManagedIdentityE2ETestsOnAzureArc'
+  displayName: 'Managed Identity E2E Tests � Azure ARC Agent'
+  dependsOn: ['BuildAndStageProjects']
+  pool:
+    name: 'MSALMSIAZUREARC'
+
+  variables:
+    runCodesignValidationInjection: false
+    Codeql.SkipTaskAutoInjection: true
+
+  steps:
+  - template: template-run-mi-e2e-azurearc.yaml
```

Comment:
Avoid using templates unless you actually need to reuse some build steps.

Reply:
the reason to add a template here is to separate the Azure ARC and IMDS related changes from the main template.